waku/common/sqlite/database.nim

```diff
@@ -127,10 +127,15 @@ template prepare*(env: Sqlite, q: string, cleanup: untyped): ptr sqlite3_stmt =
 
 proc bindParam*(s: RawStmtPtr, n: int, val: auto): cint =
   when val is openarray[byte]|seq[byte]:
+    # The constant, SQLITE_TRANSIENT, may be passed to indicate that the object is to be copied
+    #  prior to the return from sqlite3_bind_*(). The object and pointer to it must remain valid
+    #  until then. SQLite will then manage the lifetime of its private copy.
```

Comment:
I assume this means it handles the `malloc` but always `free` it again - more memory but (hopefully) not leaky memory?

Reply:
We call the [`sqlite3_finalize`](https://www.sqlite.org/c3ref/finalize.html) function later in the code to release the prepared statement memory. 

This is what the documentation says about the SQLite statement destructor:

> The application must finalize every [prepared statement](https://www.sqlite.org/c3ref/stmt.html) in order **to avoid resource leaks**.

I interpret this as deallocating any memory associated with the prepared statement. Bindings included.